includes/class-rest.php

```
@@ -241,8 +241,16 @@ private function sanitize_settings_recursively( $settings ) {
 					$sanitized[ $safe_key ] = $this->sanitize_settings_recursively( $value );
 				} elseif ( is_bool( $value ) ) {
 					$sanitized[ $safe_key ] = (bool) $value;
-				} else {
+				} elseif ( is_numeric( $value ) ) {
+					$sanitized[ $safe_key ] = (int) $value;
+				} elseif ( strpos( $safe_key, 'api_key' ) !== false || strpos( $safe_key, 'password' ) !== false ) {
+					$sanitized[ $safe_key ] = sanitize_text_field( $value );
+				} elseif ( strpos( $safe_key, 'url' ) !== false || strpos( $safe_key, 'cdn' ) !== false || strpos( $safe_key, 'host' ) !== false || strpos( $safe_key, 'origin' ) !== false ) {
```

PHPStan (2.2.5)

[warning] Parameter `#1` $haystack of function strpos expects string, string|null given.


(argument.type)